### PR TITLE
E2E: Metrics Collector Smoke Test

### PR DESCRIPTION
**Pull request by and-rewsmith**
This PR contains a simple E2E smoke test of the customer facing Metrics Collector. @pmzara added this E2E test then I did some cleanup and integrated it into the E2E pipeline.

### CODE REVIEW

**Comment by and-rewsmith** [comment]
This test passes locally but waiting for a build to confirm the pipeline integration works as expected.

File: builds/misc/images.yaml [hunk 1]
```diff
@@ -183,7 +183,7 @@ jobs:
      # Azure IoT Edge Metrics Collector
       - template: templates/image-linux.yaml
         parameters:
-          name: Azure IoT Edge Metrics Collector
+          name: Metrics Collector (customer-facing)
```
Comment (and-rewsmith) [inline builds/misc/images.yaml:186]: I chose to remove this prefix as the relevant part was getting truncated on my screen dimensions.

File: builds/e2e/templates/e2e-run.yaml [hunk 2]
```diff
@@ -31,6 +31,7 @@ steps:
     E2E_EVENT_HUB_ENDPOINT: $(TestEventHubCompatibleEndpoint)
     E2E_PREVIEW_EVENT_HUB_ENDPOINT: $(TestPreviewEventHubCompatibleEndpoint)
     E2E_IOT_HUB_CONNECTION_STRING: $(TestIotHubConnectionString)
+    E2E_IOT_HUB_RESOURCE_ID: $(TestIotHubConnectionString)
```
Comment (damonbarry) [inline builds/e2e/templates/e2e-run.yaml:34]: Should be `$(TestIotHubResourceId)`

Comment (and-rewsmith) [reply builds/e2e/templates/e2e-run.yaml:34]: Good catch. E2E test is running now so I will restart it. Luckily this yaml change doesn't require an image rebuild :)

**Review by damonbarry — DISMISSED** [review]

**Review by damonbarry — DISMISSED** [review]

**Review by veyalla — APPROVED** [review]

**Pull request merged**